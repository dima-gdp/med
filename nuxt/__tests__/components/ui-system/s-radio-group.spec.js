import { mount } from '@vue/test-utils'
import VueCompositionAPI from '@vue/composition-api'
import Vue from 'vue'
import SRadioGroup from '~/components/ui-system/s-radio/s-radio-group'

Vue.use(VueCompositionAPI)

describe('ui-system/s-radio/s-radio-group', () => {
  const RADIO_GROUP_DEFAULT_PROPS = {
    value: 0,
    column: false,
  }
  const radioGroupFactory = (propsData = {}) => {
    return mount(SRadioGroup, {
      propsData: {
        ...propsData,
      },
    })
  }

  it('s-radio-group отображается', () => {
    const wrapper = radioGroupFactory(RADIO_GROUP_DEFAULT_PROPS)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('s-radio-group классы компонента при  дефолтных значения', () => {
    const wrapper = radioGroupFactory(RADIO_GROUP_DEFAULT_PROPS)

    expect(wrapper.classes('s-radio-group')).toBe(true)
    expect(wrapper.classes('s-radio-group--column')).toBe(false)
  })

  it('s-radio-group при добавлении параметра column добавится class s-radio-group--column', () => {
    const wrapper = radioGroupFactory({
      value: 0,
      column: true,
    })
    expect(wrapper.classes('s-radio-group--column')).toBe(true)
  })

  it('s-radio-group при изменении параметра value эммитится событие  input с новым значением', async () => {
    const wrapper = radioGroupFactory({ value: 0, column: true })
    await wrapper.setProps({ value: 1, column: false })
    expect(wrapper.emitted().input).toEqual([[1]])
  })
})
