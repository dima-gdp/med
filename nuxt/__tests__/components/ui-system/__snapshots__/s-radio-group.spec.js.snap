// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-system/s-radio/s-radio-group s-radio-group отображается 1`] = `<div role="group" aria-label="radio-group" class="s-radio-group"></div>`;
